Framework/Kernel/src/InstrumentInfo.cpp

```diff
@@ -121,16 +121,62 @@ std::string InstrumentInfo::filePrefix(unsigned int runNumber) const {
 }
 
 /// Returns the name of the live listener
-const std::string &InstrumentInfo::liveListener() const {
-  return m_liveListener;
+std::string InstrumentInfo::liveListener(const std::string &name) const {
+  if (!hasLiveListenerInfo())
+    return "";
+
+  return liveListenerInfo(name).listener();
 }
 
 /** Returns the host & port to connect to for a live data stream
  *  No guarantees are given that the provided string is well-formed and valid
  *    - the caller should check this themselves
  */
-const std::string &InstrumentInfo::liveDataAddress() const {
-  return m_liveDataAddress;
+std::string InstrumentInfo::liveDataAddress(const std::string &name) const {
+  if (!hasLiveListenerInfo())
+    return "";
+
+  return liveListenerInfo(name).address();
+}
+
+/**
+ * @brief Get LiveListenerInfo for specified connection (or default)
+ * @param name Name attribute of connection to return info on
+ * @return Reference to LiveListenerInfo for specified connection
+ * @throw std::runtime_error When no listeners, or name not found
+ */
+const LiveListenerInfo &
+InstrumentInfo::liveListenerInfo(std::string name) const {
+  if (!hasLiveListenerInfo())
+    throw std::runtime_error("Attempted to access live listener for " + m_name +
+                             " instrument, which has no listeners.");
+
+  // Default to specified default connection
+  if (name.empty())
+    name = m_defaultListener;
+
+  // If no default connection specified, fallback to first connection
+  if (name.empty())
+    return m_listeners.front();
+
+  // Name specified, find requested connection
+  for (auto &listener : m_listeners) {
+    if (listener.name() == name)
```

Comment:
As discussed, when using the GUI the names will be whatever is in the Facilities.xml file, and so should always match. But when calling StartLiveData from Python, the user might use different casing. I should change these to case insensitive comparisons.